src/main/java/seedu/address/ui/LessonCard.java

```diff
@@ -0,0 +1,69 @@
+package seedu.address.ui;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import seedu.address.model.lesson.Lesson;
+
+/**
+ * An UI component that displays information of a {@code Assignment}.
+ */
+public class LessonCard extends UiPart<Region> {
+
+    private static final String FXML = "LessonListCard.fxml";
+
+    /**
+     * Note: Certain keywords such as "location" and "resources" are reserved keywords in JavaFX.
+     * As a consequence, UI elements' variable names cannot be set to such keywords
+     * or an exception will be thrown by JavaFX during runtime.
+     *
+     * @see <a href="https://github.com/se-edu/addressbook-level4/issues/336">The issue on AddressBook level 4</a>
+     */
+
+    public final Lesson lesson;
+
+    @FXML
+    private HBox cardPane;
+    @FXML
+    private Label name;
+    @FXML
+    private Label id;
+    @FXML
+    private Label deadline;
+    @FXML
+    private Label endTime;
+    @FXML
+    private Label moduleCode;
+
+    /**
+     * Creates a {@code AssignmentCode} with the given {@code Assignment} and index to display.
+     */
+    public LessonCard(Lesson lesson, int displayedIndex) {
+        super(FXML);
```

Comment:
Similar to the comment above regarding JavaDocs

Reply:
amended!